### ISSUE TITLE
object-literal-sort-keys > match-declaration-order does not handle inheritance. 

### ISSUE THREAD

**Issue by mastermatt** [palantir/tslint#3586]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.6.1
- __Running TSLint via__: CLI 

#### TypeScript code being linted

```ts
interface IFooOptional {
  a: string;
  b?: {
    foo?: string;
  };
}

interface IFooRequired extends IFooOptional {
  b: {
    foo: string;
  };
}

const x: IFooRequired = {
  a: "hello",
  b: {
    foo: "bar",
  },
};

const y: IFooRequired = {
  b: {
    foo: "bar",
  },
  a: "hello",
};
```

with `tslint.json` configuration:

```json
{
  "defaultSeverity": "error",
  "extends": [
    "tslint:recommended"
  ],
  "jsRules": {},
  "rules": {
    "object-literal-sort-keys": [true, "ignore-case", "match-declaration-order"]
  },
  "rulesDirectory": []
}
```

#### Actual behavior
Both `x` and `y` fail the rule.
> ERROR: /app/src/test.ts[15, 3]: The key 'a' is not in the same order as it is in 'IFooRequired'.
ERROR: /app/src/test.ts[25, 3]: The key 'a' is not in the same order as it is in 'IFooRequired'.

#### Expected behavior
The var `x` (line 15) should pass this rule.

**Comment by johnsoft** [comment]
#3704 is a possibly-related issue.

**Comment by La-source** [comment]
Up ?

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
